hack/push-chart.sh

```diff
@@ -35,7 +35,9 @@ readonly k8s_registry="registry.k8s.io/kueue"
 readonly semver_regex='^v([0-9]+)(\.[0-9]+){1,2}$'
 
 image_repository=${IMAGE_REPO}
-chart_version=${GIT_TAG}
+app_version=${GIT_TAG}
+# Strip leading v from version
+chart_version=${app_version/v/}
```

Comment:
It seems this would strip v from any place? This can be trick to debug if we ever add sufixes to version (currently only rc). Check of we can strip only first.

Reply:
Hey, thanks, I considered that but went with the simplest approach (Bash).
`sed` can easily be leveraged to strip leading `v`, if you believe this is really an issue.

Reply:
Would a trick like here work; https://stackoverflow.com/questions/19571391/remove-prefix-with-make?
Maybe `$(app_version:v%=%)`?

Reply:
It probably would, but IIRC, the script has the ability to figure out the version automatically based on the Git tag, so I suspect it would be preferable to have the interpolation happen within the script itself rather than injected from the caller (Makefile).

Also, having this self-contained makes testing easier (just calling the script), instead of having the implicit requirement of being called from `make`.

I'll update the MR with the `sed` version today.

Reply:
+1 on @mimowo 
Semantic Versioning allows us to add `v` in arbitrary place: https://semver.org/#is-there-a-suggested-regular-expression-regex-to-check-a-semver-string

If we replace every `v` with this script, this may cause a bug.

Reply:
> it would be preferable to have the interpolation happen within the script itself rather than injected from the caller (Makefile).

Yeah, I'm not questioning that. Making the stripping inside the bash script as currently is even better. Not sure about the syntax, but surely we can figure out something.

Reply:
Fixed it in the latest version of the change.

Examples

```shell
$ echo v1.2.3 | sed -e 's/^v//'
1.2.3
$ echo v1.2.3+v123 | sed -e 's/^v//'
1.2.3+v123
$ echo 1.2.3+v123 | sed -e 's/^v//'
1.2.3+v123
```

Reply:
CI got my back: turns out there was a magic notation to do this with Bash. Went with this solution instead.